lib/reek/smells/subclassed_from_core_class.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+
+module Reek
+  module Smells
+    # TODO: documentation
+    class SubclassedFromCoreClass < SmellDetector
+      CORE_CLASSES = [:Array, :Hash, :String]
+
+      def self.contexts
+        [:class]
+      end
+
+      # Checks +ctx+ for if it is subclasssed from a core class
+      #
+      # @return [Array<SmellWarning>]
+      def inspect(ctx)
+        _class_node, ancestor_node, _body_node = ctx.exp.to_a
+        return [] if ancestor_node.nil?
```

Comment:
I'm having trouble understanding to structure this. Could you give me some guidance? (cc @troessner)